lib/teleterm/clusters/cluster_servers.go

```diff
@@ -68,3 +71,80 @@ func (c *Cluster) GetServers(ctx context.Context) ([]Server, error) {
 
 	return results, nil
 }
+
+func (c *Cluster) GetServers(ctx context.Context, r *api.GetServersRequest) (*GetServersResponse, error) {
+	var (
+		clusterServers []types.Server
+		resp           *types.ListResourcesResponse
+		sortBy         types.SortBy
+	)
+
+	err := addMetadataToRetryableError(ctx, func() error {
+		proxyClient, err := c.clusterClient.ConnectToProxy(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer proxyClient.Close()
+
+		authClient, err := proxyClient.CurrentClusterAccessPoint(ctx)
+		// do we need to call authClient.Close() similar to proxyClient above?
+		// is there a better way to get the auth client instead of this?
```

Comment:
> do we need to call authClient.Close() similar to proxyClient above?

Yes, that's the best practice. Though it's important to close the client only after you get everything that you needed from it, see #14230.

> is there a better way to get the auth client instead of this?

The code for connecting to the proxy and the cluster from the web server is somewhat different to what tsh and Connect are doing because the web server caches the clients. The plan is to do the same in Connect sooner rather than later (#15603) but obviously we don't want to do that within this PR.

I believe we should use `ProxyClient.ConnectToCluster` as we know which cluster we're going to connect to upfront. See `Cluster.GetAllowedDatabaseUsers` below. I'm not even sure what an "access point" is but in every other place in `lib/teleterm` we use `ConnectToCluster` and AFAIR tsh usually does that as well.

Pulling the clients outside of `addMetadataToRetryableError` would also help minimize the indentation in this function. I saw some places in tsh which do similar thing with `retryWithRelogin`.

https://github.com/gravitational/teleport/blob/154ed6ea50b5fcec292d7239673f3886b2ee3a21/lib/teleterm/clusters/cluster_databases.go#L147-L185

Reply:
Sorry @avatus, I was wrong about extracting the clients outside of `addMetadataToRetryableError`, [I posted about it on Slack](https://gravitational.slack.com/archives/C0DF0TPMY/p1661944910416599). All calls to the proxy or the cluster have to be within `addMetadataToRetryableError` as the cert can expire between those calls. We should aim to have the least amount of code possible within `addMetadataToRetryableError` so that we don't unnecessarily trigger that mechanism for errors unrelated to expired certs.